src/MatrixClientPeg.js

```diff
@@ -134,9 +80,67 @@ class MatrixClient {
             console.warn("No local storage available: can't persist session!");
         }
     }
+
+    getCredentials() {
+        return [
+            this.matrixClient.baseUrl,
+            this.matrixClient.idBaseUrl,
+            this.matrixClient.credentials.userId,
+            this.matrixClient.getAccessToken(),
+            this.guestAccess.isGuest(),
+        ];
+    }
+
+    tryRestore() {
+        if (localStorage) {
+            const hs_url = localStorage.getItem("mx_hs_url");
+            const is_url = localStorage.getItem("mx_is_url") || 'https://matrix.org';
+            const access_token = localStorage.getItem("mx_access_token");
+            const user_id = localStorage.getItem("mx_user_id");
+            const guestAccess = new GuestAccess(localStorage);
+            if (access_token && user_id && hs_url) {
+                console.log("Restoring session for %s", user_id);
+                this._createClient(hs_url, is_url, user_id, access_token, guestAccess);
+            } else {
+                console.log("Session not found.");
+            }
+        }
+    }
+
+    _createClient(hs_url, is_url, user_id, access_token) {
+        var opts = {
+            baseUrl: hs_url,
+            idBaseUrl: is_url,
+            accessToken: access_token,
+            userId: user_id,
+            timelineSupport: true,
+        };
+
+        if (localStorage) {
+            opts.sessionStore = new Matrix.WebStorageSessionStore(localStorage);
+            opts.deviceId = deviceId();
+        }
+
+        this.matrixClient = Matrix.createClient(opts);
+
+        // we're going to add eventlisteners for each matrix event tile, so the
+        // potential number of event listeners is quite high.
+        this.matrixClient.setMaxListeners(500);
+
+        if (this.guestAccess) {
```

Comment:
I don't really like this. The method is pulling most inputs from its params, but guestAccess from `this`.

Possibly this whole `if.guestAccess` block could be pulled up a load? In fact, wtf even is` guestAccess` and why is it a member of `MatrixClientPeg`?